tests/stateful.lua

```diff
@@ -0,0 +1,105 @@
+-- compatibility with Lua 5.1/5.2
+local unpack = rawget(table, "unpack") or unpack
+
+function gen_stateful_iter(nret, limit)
```

Comment:
It would be a mistake to implement `drop_while` with a semantic of some `drop_if` (a reverse of `filter`). However, I pass this test with such an implementation.

```lua
stateful_methods.drop_while = method1(function(fun, gen, param, state)
    return filter(function(...)
        return not fun(...)
    end, gen, param, state)
end)
```

I propose to produce testing values that are not necessarily grow monotonically.

Reply:
Done.